snapshot/restore/local_fs_output.go

```diff
@@ -73,12 +73,28 @@ func (o *FilesystemOutput) WriteFile(ctx context.Context, relativePath string, f
 
 // CreateSymlink implements restore.Output interface.
 func (o *FilesystemOutput) CreateSymlink(ctx context.Context, relativePath string, e fs.Symlink) error {
-	log(ctx).Debugf("create symlink not implemented yet")
+	targetPath, err := e.Readlink(ctx)
+	if err != nil {
+		return errors.Wrap(err, "error reading link target")
+	}
+
+	log(ctx).Infof("CreateSymlink %v => %v", relativePath, targetPath)
+
+	path := filepath.Join(o.TargetPath, filepath.FromSlash(relativePath))
+
+	if err := os.Symlink(targetPath, path); err != nil {
```

Comment:
The only concern here is whether there should be any special handling for symlinks that point (i.e., have as target) something that is outside the tree of the snapshot. Sometimes, that's OK; sometimes it is desirable to prevent that by say, ignoring those symlinks.

Reply:
we never really look at those targets and never follow them. Perhaps this is something to change in the future.

Reply:
Correct. Kopia does not look or follow those targets.

The comment is about cases where the user does not want to restore symlinks that point to something outside the tree. This is somewhat related to the functionality provided by the `--copy-unsafe-links`, `--safe-links` and `--munge-links` flags in `rsync`.